x-pack/plugins/viz_editor/server/essql/route.ts

```diff
@@ -11,6 +11,34 @@
 import { Legacy } from 'kibana';
 import { API_PREFIX } from '../../common';
 
+export function normalizeType(type: string) {
+  const normalTypes = {
+    string: ['string', 'text', 'keyword', '_type', '_id', '_index'],
+    number: [
+      'float',
+      'half_float',
+      'scaled_float',
+      'double',
+      'integer',
+      'long',
+      'short',
+      'byte',
+      'token_count',
+      '_version',
+    ],
+    date: ['date', 'datetime'],
+    boolean: ['boolean'],
+    null: ['null'],
+  } as any;
+
+  const normalizedType = Object.keys(normalTypes).find(t => normalTypes[t].includes(type));
+
+  if (normalizedType) {
+    return normalizedType;
+  }
+  throw new Error(`Canvas does not yet support type: ${type}`);
```

Comment:
Is this really an error? There's a couple possibilities for the data types here:

- Geo types aren't listed above
- Custom types don't appear, such as if you're using a custom analyzer

So why throw an error here?

Reply:
TBH I just copied this function over from the canvas code - it makes sense though because if we can't map it back to a 'simple' `kibana_datatable` type, the downstream expression functions might not know what to do with it